server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -2100,6 +2106,138 @@ void purgeShouldProduceAFailedTaskWithVaultDeleteError() {
         }
     }
 
+    @Nested
+    class DeleteTest {
+
+        @Test
+        void deleteShouldProduceASuccessfulTaskEvenNoDeletedMessageExisted() {
+            String taskId =
+                when()
+                    .delete(BOB_DELETE_PATH)
+                    .jsonPath()
+                    .get("taskId");
+
+            given()
+                .basePath(TasksRoutes.BASE)
+            .when()
+                .get(taskId + "/await")
+            .then()
+                .body("status", is("completed"))
+                .body("taskId", is(taskId))
+                .body("type", is(DeletedMessagesVaultDeleteTask.TYPE))
+                .body("startedDate", is(notNullValue()))
+                .body("submitDate", is(notNullValue()))
+                .body("completedDate", is(notNullValue()));
+        }
+
+        @Test
+        void deleteShouldProduceASuccessfulTask() {
+            vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+            vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
+
+            String taskId =
+                when()
+                    .delete(BOB_DELETE_PATH)
+                    .jsonPath()
+                    .get("taskId");
+
+            given()
+                .basePath(TasksRoutes.BASE)
+            .when()
+                .get(taskId + "/await")
+            .then()
+                .body("status", is("completed"))
+                .body("taskId", is(taskId))
+                .body("type", is(DeletedMessagesVaultDeleteTask.TYPE))
+                .body("startedDate", is(notNullValue()))
+                .body("submitDate", is(notNullValue()))
+                .body("completedDate", is(notNullValue()));
+        }
+
```

Comment:
We don't check that:
 - delete does delete the message in the vault?
 - delete does return a task id, with the right location header?

Reply:
> delete does return a task id, with the right location header?

I can not get your point for this test case.

Reply:
See exportShouldReturnATaskCreated